mmcv/runner/fp16_utils.py

```diff
@@ -107,7 +117,11 @@ def new_func(*args, **kwargs):
                     else:
                         new_kwargs[arg_name] = arg_value
             # apply converted arguments to the decorated method
-            output = old_func(*new_args, **new_kwargs)
+            if TORCH_VERSION != 'parrots' and TORCH_VERSION >= '1.6.0':
```

Comment:
```python
with autocast(enabled=False):
      output = old_func(*new_args,  **new_kwargs)
```
will be more clear.

another question: If we use `autocast`, do we need to manually convert the `args` and `kwargs`  by invoking `cast_tensor_type`?

Reply:
fixed using with. Checked.

Also added docstring to explain why we still use cast_tensor_type